src/saveload/saveload.cpp

```diff
@@ -1767,7 +1767,7 @@ std::vector<SaveLoad> SlTableHeader(const SaveLoadTable &slt)
 			}
 
 			while (true) {
-				uint8_t type;
+				uint8_t type = 0;
```

Comment:
Here too, if you plan to "merge with rebase", this is not a fix. This really is a codechange. Fix is for user-facing issues, not for compiler-facing issues :P